File: python/조건에 맞게 수열 변환하기 2.py
def solution(arr):
    answer = 0
    old = arr
    while(True):
        new = []
        for i in old:
            if i >= 50 and i % 2 == 0:
                i = i/2
            elif i < 50 and i % 2 == 1:
                i = i*2 + 1
            new.append(int(i))
        if old == new:
            break
        else:
            old = new
            answer += 1
    return answer

File: java/조건에 맞게 수열 변환하기 2.java
class Solution {
    public int solution(int[] arr) {
        int answer = 0;
        int cnt = 0;
        while (true) {
            cnt = 0;
            for (int i = 0; i < arr.length; i++) {
                if (arr[i] >= 50 && arr[i] % 2 == 0) {
                    arr[i] /= 2;
                    cnt++;
                }else if (arr[i] < 50 && arr[i] % 2 == 1) {
                    arr[i] = arr[i] * 2 + 1;
                    cnt++;
                }
            }
            if (cnt == 0) {
                break;
            }
            answer++;
        }
        return answer;
    }
}